tests/providers/amazon/aws/utils/eks_test_utils.py

```diff
@@ -164,10 +164,14 @@ def region_matches_partition(region: str, partition: str) -> bool:
         ("us-gov-iso-b-", "aws-iso-b"),
     ]
 
-    for prefix, expected_partition in valid_matches:
-        if region.startswith(prefix):
-            return partition == expected_partition
-    return partition == "aws"
+    return next(
+        (
+            partition == expected_partition
+            for prefix, expected_partition in valid_matches
+            if region.startswith(prefix)
+        ),
+        partition == "aws",
+    )
```

Comment:
Nit: Defining a generator inline and calling next with the default is fun Python but I think it is a bit unnecessary, it's turning 4 lines of very readable code into 8 lines of obtuse Python.